src/sparseml/pytorch/sparsification/distillation/modifier_per_layer.py

```diff
@@ -208,28 +211,64 @@ def epsilon(self, value: float):
 
     @ModifierProp()
     def student_layer_names(self) -> List[str]:
+        """
+        :return: List of layer names to distill.
+        """
         return self._student_layer_names
 
     @student_layer_names.setter
     def student_layer_names(self, value: List[str]):
+        """
+        :param value: List of layer names to distill.
+            *Must be same length as teacher_layer_names*
+        """
         self._student_layer_names = value
 
     @ModifierProp()
     def teacher_layer_names(self) -> List[str]:
+        """
+        :return: List of layer names to distill
+        """
         return self._teacher_layer_names
 
     @teacher_layer_names.setter
     def teacher_layer_names(self, value: List[str]):
+        """
+        :param value: List of layer names to distill.
+            *Must be same length as student_layer_names*
+        """
         self._teacher_layer_names = value
 
     @ModifierProp()
     def project_features(self) -> bool:
+        """
+        :return: Whether to perform projection
+            between student and teacher layers
+        """
         return self._project_features
 
     @project_features.setter
     def project_features(self, value: bool):
+        """
+        :param value: Whether to perform projection
+            between student and teacher layers
+        """
         self._project_features = value
 
+    def state_dict(self) -> Dict[str, Dict]:
+        state = super().state_dict()
+        if self.project_features:
+            state[_DISTILL_PARAM_GROUP_KEY] = {
+                name: p.state_dict() for name, p in self._projection.items()
+            }
+        return state
+
+    def load_state_dict(self, state_dict: Dict[str, Dict], strict: bool = True):
+        if self.project_features:
+            # save until self._projection is actually initialized after forward
+            self._loaded_projection = state_dict.pop(_DISTILL_PARAM_GROUP_KEY)
```

Comment:
why are we popping? this would modify the passed state dict in-place, no?

Reply:
to handle cases where the key is not present, we could use `.get` instead to avoid mutating

Reply:
let's do that